bitcoin/src/network/constants.rs

```diff
@@ -144,7 +166,7 @@ impl FromStr for Network {
             "testnet" => Testnet,
             "signet" => Signet,
             "regtest" => Regtest,
-            _ => return Err(ParseNetworkError(s.to_owned())),
+            _ => return Network::from_core_arg(s),
         };
         Ok(network)
```

Comment:
Perhaps this would be cleaner
```rust
    fn from_str(s: &str) -> Result<Self, Self::Err> {
        // Some comment about why "bitcoin" is used as well as "mainnet".
        if s == "bitcoin" {
            Ok(Network::Bitcoin)
        } else {
            Network::from_core_arg(s)
        }
    }
```

Reply:
Agreed modulo `testnet` is agreed upon as an option for `from_core_args`